### PR TITLE
add dependabot configuration

### PR DESCRIPTION
**Pull request by osterman**
## what
* add dependabot configuration

## why
* automatic module PRs

### CODE REVIEW

File: templates/.github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    schedule:
+      interval: "daily"
+    labels:
+      - "terraform-submodules"
+
+  # Maintain dependencies for Terraform Examples
+  - package-ecosystem: "terraform"
+    rebase-strategy: "auto"
+    open-pull-requests-limit: 5
+    directory: "/examples/complete/"
```
Comment (nitrocode) [inline templates/.github/dependabot.yml:40]: why not all examples ?

**Comment by lorengordon** [comment]
Heads up that dependabot upstream _still_ [does not support HCL2](https://github.com/dependabot/dependabot-core/pull/1388). There are a couple options that function as a separate github action, though...

* https://github.com/patrickjahns/dependabot-terraform-action
* https://github.com/plus3it/dependabot-terraform-action (our fork, mostly shrinking the dependencies, iirc)

Something like this is what we use:
```
name: Update terraform dependencies

on:
  schedule:
    - cron:  '0 6 * * *'

jobs:
  dependabot-terraform:
    runs-on: ubuntu-latest
    steps:
      - name: update terraform dependencies
        uses: plus3it/dependabot-terraform-action@p3-implementation
        with:
          github_dependency_token: ${{ secrets.GH_READONLY_TOKEN }}
```

File: templates/.github/dependabot.yml (hunk 1 above)
Comment (osterman) [reply templates/.github/dependabot.yml:40]: Unfortunately, I don't think there's a way to tell it to go recursively. I would need to know what all examples are upfront or generate the file (which is out-of-scope for this PR).

**Comment by osterman** [comment]
Ah... I remember someone saying that on office hours. I could have sworn it was working for us, but I looked back and the PR history and don't see any coming in anymore for terraform, so I guess that's why.

@lorengordon does this action handle the rebasing?

**Comment by lorengordon** [comment]
> @lorengordon does this action handle the rebasing?

I don't believe so, no... That's part of the Dependabot SaaS offering, not the packaged container. Maybe there's another github action that could do that? Or maybe mergify?

**Comment by Nuru** [comment]
Looks like Renovate is a better option. We can revisit/reopen this later if we change our minds.

**Pull request closed without merging**